backend/src/achievements/dto/response-achievement.dto.ts

```diff
@@ -0,0 +1,4 @@
+import { OmitType } from '@nestjs/swagger';
+import { AchievementDto } from './achievement.dto';
+
+export class ResponseAchievementDto extends OmitType(AchievementDto, ['id']) {}
```

Comment:
```suggestion
export class ResponseAchievementDto extends OmitType(AchievementDto, []) {}
```
I would include the `id` of the entities in the response in case the client:
- needs to make requests that reference it later
- receives a reference to the achievement by `id`